apps/emqx_management/src/emqx_mgmt_api_pubsub.erl

```diff
@@ -71,19 +71,10 @@ subscribe(_Bindings, Params) ->
 
 publish(_Bindings, Params) ->
     logger:debug("API publish Params:~p", [Params]),
-    {ClientId, Topic, Qos, Retain, Payload, UserProps} = parse_publish_params(Params),
-    case do_publish(ClientId, Topic, Qos, Retain, Payload, UserProps) of
-        {ok, MsgIds} ->
-            case proplists:get_value(<<"return">>, Params, undefined) of
-                undefined -> minirest:return(ok);
-                _Val ->
-                    case proplists:get_value(<<"topics">>, Params, undefined) of
-                        undefined -> minirest:return({ok, #{msgid => lists:last(MsgIds)}});
-                        _ -> minirest:return({ok, #{msgids => MsgIds}})
-                    end
-            end;
-        Result ->
-            minirest:return(Result)
+    try parse_publish_params(Params) of
+        Result -> do_publish(Params, Result)
```

Comment:
Result = parse_publish_params(Params),
do_publish(Params, Result)
remove try of, because case only result

Reply:
Error maybe happend in `parse_publish_params/1`.
Api func `publish/2` need to return `BAD_REQUEST` by catched error.